cardano-db-sync/src/Cardano/DbSync/Plugin/Epoch.hs

```diff
@@ -71,55 +71,57 @@ epochPluginOnStartup trce = do
-      -- the Ouroboros Classic era.
-      pure $ Right ()
-
-    Ledger.ABOBBlock blk -> do
-      slotsPerEpoch <- liftIO $ readIORef slotsPerEpochVar
-      mLatestCachedEpoch <- liftIO $ readIORef latestCachedEpochVar
-      chainTipEpoch <- liftIO $ readIORef latestChainTipEpochVar
-      let epochNum = Byron.epochNumber blk slotsPerEpoch
-          lastCachedEpoch = fromMaybe 0 mLatestCachedEpoch
-
-
-
-      if  | epochNum == chainTipEpoch && lastCachedEpoch == chainTipEpoch ->
-              if withOrigin 0 unBlockNo (getTipBlockNo tip) - Byron.blockNumber blk < 15
-                then -- Following the chain very closely.
-                     updateEpochNum epochNum trce
-                else pure $ Right ()
-
-          | epochNum > 0 && mLatestCachedEpoch == Nothing ->
-              updateEpochNum 0 trce
-          | epochNum >= lastCachedEpoch + 2 ->
-              updateEpochNum (lastCachedEpoch + 1) trce
-          | epochNum == chainTipEpoch && lastCachedEpoch < chainTipEpoch ->
-              updateEpochNum (lastCachedEpoch + 1) trce
-          | epochNum > chainTipEpoch ->
-              -- Must just have started a new epoch, so call this which will
-              -- update chainTipEpoch.
-              updateEpochNum epochNum trce
-          | otherwise ->
-              pure $ Right ()
-
-epochPluginRollbackBlock :: Trace IO Text -> Point ByronBlock -> IO (Either DbSyncNodeError ())
-epochPluginRollbackBlock _trce point =
-    case Byron.pointToSlotHash point of
-      Nothing -> pure $ Right ()
-      Just (slot, _hash) -> DB.runDbAction Nothing $ action (Ledger.unSlotNumber slot)
-  where
-    action :: MonadIO m => Word64 -> ReaderT SqlBackend m (Either DbSyncNodeError ())
-    action slot = do
-      slotsPerEpoch <- liftIO $ readIORef slotsPerEpochVar
-      delete . from $ \ epoch ->
-        where_ (epoch ^. EpochNo >=. val (slot `div` slotsPerEpoch))
-      transactionSaveWithIsolation Serializable
-      pure $ Right ()
+epochPluginInsertBlock :: Trace IO Text -> CardanoBlockTip -> ReaderT SqlBackend (LoggingT IO) (Either DbSyncNodeError ())
+epochPluginInsertBlock trce blkTip = do
+  slotsPerEpoch <- liftIO $ readIORef slotsPerEpochVar
+  case blkTip of
+    ByronBlockTip bblk tip ->
+      case byronBlockRaw bblk of
+        Byron.ABOBBoundary _ ->
+          -- For the OBFT era there are no boundary blocks so we ignore them even in
+          -- the Ouroboros Classic era.
+          pure $ Right ()
+
+        Byron.ABOBBlock blk ->
+          insertBlock trce (Byron.epochNumber blk slotsPerEpoch, Byron.blockNumber blk) (tipBlockNo tip)
+    ShelleyBlockTip sblk tip ->
+      insertBlock trce (Shelley.epochNumber sblk slotsPerEpoch, Shelley.blockNumber sblk) (tipBlockNo tip)
+
+-- Nothing to be done here.
+-- Rollback will take place in the Default plugin and the epoch table will be recalculated.
+epochPluginRollbackBlock :: Trace IO Text -> CardanoPoint -> IO (Either DbSyncNodeError ())
```

Comment:
Why do we need this here anyway?

Reply:
The epoch plugin was a proof-of-concept thing. It is there for documentation.